src/Resolvers/Microsoft.DotNet.NativeWrapper/Interop.cs

```diff
@@ -12,36 +13,28 @@ namespace Microsoft.DotNet.NativeWrapper
     public static partial class Interop
     {
         public static readonly bool RunningOnWindows = RuntimeInformation.IsOSPlatform(OSPlatform.Windows);
+        static readonly string HostFxrPath;
 
         static Interop()
         {
-            if (RunningOnWindows)
-            {
-                PreloadWindowsLibrary("hostfxr.dll");
-            }
+            HostFxrPath = (string)AppContext.GetData("HOSTFXR_PATH");
```

Comment:
Check that `HostFxrPath`? I don't know if the code here can assume that the hosting bits launching it have the HOSTFXR_PATH change.

Reply:
Technically it should be able to make that assumption (SDK is bound to a version of runtime), but I agree it would be a good idea to somehow handle the error more gracefully.